pkg/bloomgateway/processor.go

```diff
@@ -136,15 +127,23 @@ func (p *processor) processBlocks(ctx context.Context, bqs []*bloomshipper.Close
 			return errors.Errorf("block and querier bounds differ: %s vs %s", block.ref.Bounds, bq.Bounds)
 		}
 
-		err := p.processBlock(ctx, bq.BlockQuerier, block.tasks)
+		err := p.processBlock(ctx, bq, block.tasks)
 		if err != nil {
 			return errors.Wrap(err, "processing block")
 		}
 		return nil
 	})
 }
 
-func (p *processor) processBlock(_ context.Context, blockQuerier *v1.BlockQuerier, tasks []Task) error {
+func (p *processor) processBlock(_ context.Context, bq *bloomshipper.CloseableBlockQuerier, tasks []Task) (err error) {
+	defer func() {
+		var errs multierror.MultiError
+		errs.Add(err)
+		errs.Add(bq.Close())
+		err = errs.Err()
```

Comment:
That looks wrong, since on line 141 you add `err` to the multi-error.

Reply:
That's capturing the `err` from the function signature and adding it to the multi-error (which can now have a main error from the body of the function call and/or an error from cleanup.